enclave/src/main/java/tech/pegasys/pantheon/enclave/Enclave.java

```diff
@@ -73,11 +77,18 @@ private Request buildPostRequest(
   }
 
   private <T> T executePost(final Request request, final Class<T> responseType) throws IOException {
-    try (Response response = client.newCall(request).execute()) {
-      return objectMapper.readValue(response.body().string(), responseType);
-    } catch (IOException e) {
-      LOG.error("Enclave failed to execute ", request);
-      throw new IOException("Enclave failed to execute post", e);
+    for (int i = 0; i < RETRIES; i++) {
+      try (Response response = client.newCall(request).execute()) {
+        return objectMapper.readValue(response.body().string(), responseType);
+      } catch (IOException e) {
+        LOG.error("Enclave failed to execute {}", request);
+      }
+      try {
+        TimeUnit.SECONDS.sleep(POLL_TIME_SECONDS);
```

Comment:
Is it safe to sleep the executing thread here?

Reply:
Yes - this was added because of a flaky test. I have since fixed the test.